src/model/GeneratorWindTurbine.hpp

```diff
@@ -0,0 +1,201 @@
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_GENERATORWINDTURBINE_HPP
+#define MODEL_GENERATORWINDTURBINE_HPP
+
+#include "ModelAPI.hpp"
+#include "Generator.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+  namespace detail {
+
+    class GeneratorWindTurbine_Impl;
+
+  }  // namespace detail
+
+  /** GeneratorWindTurbine is a Generator that wraps the OpenStudio IDD object 'OS:Generator:WindTurbine'. */
+  class MODEL_API GeneratorWindTurbine : public Generator
+  {
+   public:
+    /** @name Constructors and Destructors */
+    //@{
+
+    explicit GeneratorWindTurbine(const Model& model);
+
+    virtual ~GeneratorWindTurbine() {}
+
+    //@}
+
+    static IddObjectType iddObjectType();
+
+    /** @name Getters */
+    //@{
+
+    std::string rotorType() const;
+
+    std::string powerControl() const;
+
+    double ratedRotorSpeed() const;
+
+    double rotorDiameter() const;
+
+    double overallHeight() const;
+
+    int numberofBlades() const;
+
+    double ratedPower() const;
+
+    double ratedWindSpeed() const;
+
+    double cutInWindSpeed() const;
+
+    double cutOutWindSpeed() const;
+
+    double fractionSystemEfficiency() const;
+
+    double maximumTipSpeedRatio() const;
+
+    double maximumPowerCoefficient() const;
+
+    boost::optional<double> annualLocalAverageWindSpeed() const;
+
+    double heightforLocalAverageWindSpeed() const;
+
+    boost::optional<double> bladeChordArea() const;
+
+    double bladeDragCoefficient() const;
+
+    double bladeLiftCoefficient() const;
+
+    double powerCoefficientC1() const;
+
+    double powerCoefficientC2() const;
+
+    double powerCoefficientC3() const;
+
+    double powerCoefficientC4() const;
+
+    double powerCoefficientC5() const;
+
+    double powerCoefficientC6() const;
+
+    //@}
+    /** @name Setters */
+    //@{
+
+    bool setAvailabilitySchedule(Schedule& schedule);
+
+    void resetAvailabilitySchedule();
```

Comment:
Remove

Reply:
I would **still** remove this even though the getter will be optional per Generator.hpp.